### PR TITLE
add more EM datasets

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: bolts/fair/data/datamodules/tabular/admissions.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+class AdmissionsDataModule(TabularDataModule):
+    """Data Module for the Admissions Dataset."""
+
+    @parsable
+    def __init__(
+        self,
+        sens_feat: AdmissionsSplits = "Gender",
+        disc_feats_only: bool = False,
+        # Below are super vars. Not doing *args **kwargs due to this being parsable
```
Comment (thomkeh) [inline bolts/fair/data/datamodules/tabular/admissions.py:23]: this is one annoyance that dataclasses solve, but we probably can't combine DataModules and dataclasses

File: bolts/fair/data/datamodules/tabular/compas.py [hunk 2]
```diff
@@ -9,6 +14,40 @@
 class CompasDataModule(TabularDataModule):
     """COMPAS Dataset."""
 
+    @parsable
+    def __init__(
+        self,
+        sens_feat: Literal["Sex", "Race", "Race-Sex"] = "Sex",
```
Comment (thomkeh) [inline bolts/fair/data/datamodules/tabular/compas.py:20]: `Literal` isn't "parsable" by hydra, right?

File: bolts/data/datamodules/base.py [hunk 3]
```diff
@@ -58,7 +58,8 @@ def logger(self) -> logging.Logger:
         return self._logger
 
     def log(self, msg: str) -> None:
-        self._logger.info(msg)
+        if self._logger is not None:
+            self._logger.info(msg)
```
Comment (MylesBartlett) [inline bolts/data/datamodules/base.py:62]:
```suggestion
     self.logger.info(msg)
```

File: bolts/data/datasets/base.py [hunk 4]
```diff
@@ -49,7 +49,8 @@ def logger(self) -> logging.Logger:
         return self._logger
 
     def log(self, msg: str) -> None:
-        self._logger.info(msg)
+        if self._logger is not None:
+            self._logger.info(msg)
```
Comment (MylesBartlett) [inline bolts/data/datasets/base.py:53]:
```suggestion
       self.logger.info(msg)
```

File: bolts/fair/data/datamodules/tabular/adult.py [hunk 5]
```diff
@@ -9,6 +15,49 @@
...
     """UCI Adult Income Dataset."""
 
+    @parsable
+    def __init__(
+        self,
+        bin_nationality: bool = False,
+        sens_feat: AdultSplits = "Sex",
+        bin_race: bool = False,
+        discrete_feats_only: bool = False,
+        # Below are super vars. NOt doing *args **kwargs due to this being parsable
```
Comment (MylesBartlett) [inline bolts/fair/data/datamodules/tabular/adult.py:25]:
```suggestion
        # Below are super vars. Not doing *args **kwargs due to this being parsable
```

File: bolts/fair/data/datamodules/tabular/compas.py (hunk 2 above)
Comment (MylesBartlett) [reply bolts/fair/data/datamodules/tabular/compas.py:20]: Yes, this needs to be an enum

File: bolts/fair/data/datamodules/tabular/credit.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+__all__ = ["CreditDataModule"]
+
+
+class CreditDataModule(TabularDataModule):
+    """Data Module for the Credit Dataset."""
+
+    @parsable
+    def __init__(
+        self,
+        sens_feat: Literal["Sex"] = "Sex",
```
Comment (MylesBartlett) [inline bolts/fair/data/datamodules/tabular/credit.py:20]: as above, this needs to be an enum, not a string

File: bolts/fair/data/datamodules/tabular/crime.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+__all__ = ["CrimeDataModule"]
+
+
+class CrimeDataModule(TabularDataModule):
+    """Data Module for the Crime Dataset."""
+
+    @parsable
+    def __init__(
+        self,
+        sens_feat: Literal["Race-Binary"] = "Race-Binary",
```
Comment (MylesBartlett) [inline bolts/fair/data/datamodules/tabular/crime.py:20]: and again

File: bolts/fair/data/datamodules/tabular/compas.py [hunk 8]
```diff
@@ -11,13 +12,19 @@
...
+
+
 class CompasDataModule(TabularDataModule):
     """COMPAS Dataset."""
 
     @parsable
     def __init__(
         self,
-        sens_feat: Literal["Sex", "Race", "Race-Sex"] = "Sex",
+        sens_feat: CompasSens.sex = CompasSens.sex,
```
Comment (MylesBartlett) [inline bolts/fair/data/datamodules/tabular/compas.py:27]:
```suggestion
        sens_feat: CompasSens = CompasSens.sex,
```

File: bolts/data/datasets/base.py (hunk 4 above)
Comment (olliethomas) [reply bolts/data/datasets/base.py:53]: can this be chained like this? `self.property.function_of_returned_value`?

Comment (MylesBartlett) [reply bolts/data/datasets/base.py:53]: I don't see why not

**Pull request merged**